src/main/java/uk/gov/companieshouse/company_appointments/model/transformer/DeltaItemLinkTypesTransformer.java

```diff
@@ -0,0 +1,39 @@
+package uk.gov.companieshouse.company_appointments.model.transformer;
+
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.appointment.ItemLinkTypes;
+import uk.gov.companieshouse.api.appointment.OfficerLinkTypes;
+import uk.gov.companieshouse.company_appointments.exception.FailedToTransformException;
+import uk.gov.companieshouse.company_appointments.model.data.DeltaItemLinkTypes;
+import uk.gov.companieshouse.company_appointments.model.data.DeltaOfficerLinkTypes;
+
+@Component
+public class DeltaItemLinkTypesTransformer implements
+        Transformative<ItemLinkTypes, DeltaItemLinkTypes> {
+
+    @Override
+    public DeltaItemLinkTypes factory() {
+        return new DeltaItemLinkTypes();
+    }
+
+    @Override
+    public DeltaItemLinkTypes transform(ItemLinkTypes source, DeltaItemLinkTypes entity)
+            throws FailedToTransformException {
+
+        try {
+            entity.setSelf(source.getSelf());
+
+            OfficerLinkTypes officerLinkTypes = source.getOfficer();
+            if (officerLinkTypes != null) {
```

Comment:
Will this always be true because the value of source.getOfficer() is annotated with a @NotNull annotation?

Reply:
I am not sure we have the validation enabled on at runtime so it is probably best to keep the check